### PR TITLE
Fix broken duplicate test name detection in TestParser.parse_all()

### LINKED ISSUE #874: Bug: Duplicate test name detection in TestParser.parse_all() is broken
## Summary

`TestParser.parse_all()` in `src/cloudai/test_parser.py` has a duplicate name check that **never fires** due to a type mismatch. When two TOML test configs share the same `name` field, they are silently loaded without error, causing config collisions at runtime.

## Root Cause

Line 64 checks `if obj_name in objects`, where `obj_name` is a `str` and `objects` is a `List[TestDefinition]` (Pydantic `BaseModel` subclasses):

```python
objects: List[Any] = []
for f in self.test_tomls:
    ...
    parsed_object = self._parse_data(data)
    obj_name: str = parsed_object.name
    if obj_name in objects:                    # bug: str in List[BaseModel]
        raise ValueError(f"Duplicate name found: {obj_name}")
    objects.append(parsed_object)
```

Python's `in` operator on a list does `obj_name == item` for each item:

1. `str.__eq__(BaseModel)` returns `NotImplemented` (str only compares with str)
2. `BaseModel.__eq__(str)` returns `NotImplemented` (Pydantic checks `isinstance(other, BaseModel)`)
3. Both returned `NotImplemented`, Python falls back to identity (`is`) which is always `False`

The check passes silently for every element, so duplicates are never detected.

## How to Reproduce

**Real-world scenario:** In `conf/staging/nemo_perf/m-bridge/test/h100/`, two files had the same name:
- `dse_qwen_30b_a3b.toml` (DSE config) with `name = "dse_qwen_30b_a3b"`
- `qwen_30b_a3b.toml` (single-point copy) with `name = "dse_qwen_30b_a3b"` (forgot to rename)

Both loaded without error. The duplicate caused a silent config collision.

**Unit test reproduction** (add to `tests/test_parser.py`):

```python
import toml
from unittest.mock import Mock, patch
from cloudai.core import TestParser

class TestParseAllDuplicateDetection:
    __test__ = True

    @staticmethod
    def _write_test_toml(path, name):
        path.write_text(toml.dumps({"name": name, "description": "test"}))

    @staticmethod
    def _make_fake_parsed(name):
        obj = Mock()
        obj.name = name
        return obj

    def test_duplicate_name_across_files_raises(self, tmp_path):
        """Two files with the same name field should raise ValueError."""
        test_dir = tmp_path / "tests"
        test_dir.mkdir()
        dse_toml = test_dir / "dse_qwen_30b_a3b.toml"
        single_toml = test_dir / "qwen_30b_a3b.toml"
        self._write_test_toml(dse_toml, "dse_qwen_30b_a3b")
        self._write_test_toml(single_toml, "dse_qwen_30b_a3b")

        parser = TestParser([dse_toml, single_toml], None)
        fake = self._make_fake_parsed("dse_qwen_30b_a3b")

        with patch.object(parser, "_parse_data", return_value=fake):
            with pytest.raises(ValueError, match="dse_qwen_30b_a3b"):
                parser.parse_all()

    def test_unique_names_no_error(self, tmp_path):
        """Files with distinct names should parse without error."""
        test_dir = tmp_path / "tests"
        test_dir.mkdir()
        toml_a = test_dir / "test_a.toml"
        toml_b = test_dir / "test_b.toml"
        self._write_test_toml(toml_a, "test_a")
        self._write_test_toml(toml_b, "test_b")

        parser = TestParser([toml_a, toml_b], None)

        with patch.object(
            parser,
            "_parse_data",
            side_effect=[self._make_fake_parsed("test_a"), self._make_fake_parsed("test_b")],
        ):
            results = parser.parse_all()
            assert len(results) == 2
```

Running against current `main`:
```
FAILED  test_duplicate_name_across_files_raises - Failed: DID NOT RAISE ValueError
PASSED  test_unique_names_no_error
```

## Suggested Fix

Track seen names in a `dict[str, Path]` instead of checking `str in List[BaseModel]`:

```python
seen_names: dict[str, Path] = {}
...
if obj_name in seen_names:
    raise ValueError(
        f"Duplicate test name '{obj_name}' found in:\n"
        f"  - {seen_names[obj_name]}\n"
        f"  - {f}"
    )
seen_names[obj_name] = f
```

This also improves the error message by listing both conflicting file paths.

### PR DESCRIPTION
**Pull request by rutayan-nv**
## Summary

- Fix the duplicate test name check in `TestParser.parse_all()` which never fires due to a type mismatch (`str in List[BaseModel]` always returns `False`)
- Replace with a `dict[str, Path]` lookup that correctly detects duplicates and reports both conflicting file paths
- Add 3 unit tests covering duplicate detection, unique names, and single-file parsing

## Test plan

- [x] `test_duplicate_name_across_files_raises` — confirms duplicates now raise `ValueError`
- [x] `test_unique_names_no_error` — confirms distinct names parse without error
- [x] `test_single_file_no_error` — confirms single file parses without error
- [x] All 14 tests in `tests/test_parser.py` pass

Fixes https://github.com/NVIDIA/cloudai/issues/874
https://redmine.mellanox.com/issues/4979846

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -204,3 +205,76 @@ def test_malformed_config_reported(self, scenario_report: str):
...
+
+        parser = TestParser([dse_toml, single_toml], None)  # type: ignore
+        fake = self._make_fake_parsed("dse_qwen_30b_a3b")
+
+        with patch.object(parser, "_parse_data", return_value=fake):
+            with pytest.raises(ValueError, match="dse_qwen_30b_a3b") as exc_info:
+                parser.parse_all()
+            msg = str(exc_info.value)
+            assert str(dse_toml) in msg
+            assert str(single_toml) in msg
```
[the remark this answers is not in the document]
Comment (rutayan-nv) [reply tests/test_parser.py:248]: False positive — lines 246-248 are indented at 12 spaces, same level as the `with pytest.raises` statement, so they execute **after** the context manager exits with the exception captured in `exc_info`. They are outside `pytest.raises` but inside `patch.object`. No change needed.

**Comment by podkidyshev** [comment]
please fix CI. otherwise looks good

**Comment by podkidyshev** [comment]
please also link internal ticket if there's any

**Comment by rutayan-nv** [comment]
> please also link internal ticket if there's any

The github issue is part of the description. I will add the redmine as well.

**Comment by rutayan-nv** [comment]
@podkidyshev pushed fixes for the CI failures. They were copyright header failures mostly.

**Review by podkidyshev — APPROVED** [review]
